### ISSUE TITLE
useless_let_if_seq should not emit a warning if multiple assignments happen

### ISSUE THREAD

**Issue by DaanVandenBosch** [rust-lang-nursery/rust-clippy#2749]
I have code that looks like this:

```rust
let img_x: u32;
let img_y: u32;
let pattern_x: u32;
let pattern_y: u32;

if forward {
    img_x = start_x + scan_x;
    img_y = start_y + scan_y;
    pattern_x = scan_x;
    pattern_y = scan_y;
} else {
    img_x = start_x - scan_x;
    img_y = start_y - scan_y;
    pattern_x = scan_width - scan_x - 1;
    pattern_y = scan_height - scan_y - 1;
}
```
Am I right in thinking that this should be ok with clippy or is there actually a more idiomatic way of doing something like this?

**Comment by jdnewman85** [comment]
(New to rust, so I may be wrong.)

I think something like this:
`
let (img_x, img_y, pattern_x, pattern_y) = if forward {
    start_x + scan_x,
    start_y + scan_y,
    scan_x,
    scan_y
} else {
    start_x - scan_x,
    start_y - scan_y,
    scan_width - scan_x - 1,
    scan_height - scan_y - 1
}
`

There is a bit less typing, but primarily combining the declaration with initialization reduces the chances of creating a bug later when editing. If many more variables are added, then it'd probably be better to wrap them in something.
